### PR TITLE
CA-72 feat: add project header to cost estimation landing screen

### PR DESCRIPTION
**Pull request by SimonDerejeW**
This PR introduces the addition of a project header app bar to the cost estimation landing screen along with the relevant widget tests.

### Package Changes
- Added package for mocking `NetworkImage` for testing widgets that contain it.

### Other Changes
- Added the dashboard and estimation modules to app module.

Note: The automated code review hasn't been run because the corresponding PR hasn't been merged at the time of writing. It will be added as soon as it goes through.

# AI Review

| Issue Name | Description | Status | Notes |
|------------|-------------|--------|-------|
| Hardcoded Project Name | The project name "My project" is hardcoded in CostEstimationLandingPage  | 📋 | Project repository is not yet implemented and that feature is out of the PR scope. |
| Hardcoded Fallback URL | The fallback avatar URL is hardcoded in ProjectHeaderAppBar  | ✅ | The issue of choosing a fallback URL for the user avatar is not the responsibility of the cost estimation module. The temporary fallback URL has been removed. |
| Empty Body Content | The body Column in CostEstimationLandingPage is empty | 📋 | This is a separate youtrack issue. |

### CODE REVIEW

**Comment by SimonDerejeW** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/ripplearc/construculator-app/96?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#96** <a href="https://app.graphite.dev/github/pr/ripplearc/construculator-app/96?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/ripplearc/construculator-app/96?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#95** <a href="https://app.graphite.dev/github/pr/ripplearc/construculator-app/95?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#92** <a href="https://app.graphite.dev/github/pr/ripplearc/construculator-app/92?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#93](https://github.com/ripplearc/construculator-app/pull/93) <a href="https://app.graphite.dev/github/pr/ripplearc/construculator-app/93?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#91** <a href="https://app.graphite.dev/github/pr/ripplearc/construculator-app/91?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by SimonDerejeW** [comment]
#runcheck

**Review by bekiblue — APPROVED** [review]
LGTM!

File: lib/features/dashboard/presentation/pages/dashboard_page.dart [hunk 1]
```diff
@@ -97,6 +98,13 @@ class _DashboardPageState extends State<DashboardPage> {
                 label: 'Logout',
                 centerAlign: true,
               ),
+              CoreButton(
+                onPressed: () {
+                  _router.navigate(fullEstimationLandingRoute);
+                },
+                label: 'Cost Estimation',
```
Comment (ripplearcgit) [inline lib/features/dashboard/presentation/pages/dashboard_page.dart]: Use `location`.

Please add a story to write how to add `localization` in the `Best Engineering Best Practice`.

File: lib/features/estimation/presentation/widgets/project_header_app_bar.dart [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                  height: 24,
+                  child: CoreIconWidget(
+                    icon: CoreIconData.svg('assets/icons/bell_icon.svg'),
+                    size: 24,
+                  ),
+                ),
+              ),
+              Container(
+                margin: const EdgeInsets.only(right: 16, left: 8),
+                child: CircleAvatar(
```
Comment (ripplearcgit) [inline lib/features/estimation/presentation/widgets/project_header_app_bar.dart]: is this a CoreUI component?

File: lib/libraries/auth/testing/fake_auth_manager.dart [hunk 3]
```diff
@@ -317,9 +317,13 @@ class FakeAuthManager implements AuthManager {
...
 
-    final result = await _authRepository.getUserProfile(credentialId);
-    _authNotifier.emitUserProfileChanged(result);
-    return AuthResult.success(result);
+    try {
+      final result = await _authRepository.getUserProfile(credentialId);
+      _authNotifier.emitUserProfileChanged(result);
+      return AuthResult.success(result);
+    } catch (e) {
+      rethrow;
```
Comment (ripplearcgit) [inline lib/libraries/auth/testing/fake_auth_manager.dart]: Should put logs around these critical error?

File: test/units/bloc/auth/auth_bloc_test.dart [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+        expect: () => [
+          const AuthLoadInProgress(),
+          isA<AuthLoadSuccess>()
+              .having((s) => s.user!.id, 'user.id', testUserId),
+          const AuthLoadUnauthenticated(),
+        ],
+      );
+    });
+
+    group('Edge Cases', () {
```
Comment (ripplearcgit) [inline test/units/bloc/auth/auth_bloc_test.dart]: too generic and broad

File: test/units/bloc/auth/auth_bloc_test.dart [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+    late FakeAppRouter fakeRouter;
+    late Clock clock;
+    late FakeSupabaseWrapper fakeSupabase;
+
+    const testUserId = 'test-user-123';
+    const testEmail = 'test@example.com';
+    const testAvatarUrl = 'https://example.com/avatar.jpg';
+
+    setUp(() {
+      // Create fake dependencies directly instead of using Modular
```
[the remark this answers is not in the document]
Comment (SimonDerejeW) [reply test/units/bloc/auth/auth_bloc_test.dart]: I faced significant challenges when trying to use Modular the same way it was used in the rest of the tests. This tests interacts with the auth manager and the calls to getUserProfile were not being registered when using modular. This may be an issue that warrants further investigation, but using Modular (as with the rest of the bloc tests in the same directory) did not work.

File: test/widgets/estimation/project_header_app_bar_test.dart [hunk 6]
```diff
@@ -2,266 +2,182 @@ import 'package:construculator/features/estimation/presentation/widgets/project_
...
+      // Check for SizedBox between project name and arrow
+      expect(find.byType(SizedBox), findsWidgets);
 
-        // Check for Container with margin around avatar
-        final containers = find.byType(Container);
-        expect(containers, findsWidgets);
-      });
+      // Check for Container with margin around avatar
+      final containers = find.byType(Container);
+      expect(containers, findsWidgets);
```
Comment (ripplearcgit) [inline test/widgets/estimation/project_header_app_bar_test.dart]: Are we testing technical details here :-(

File: pubspec.lock [hunk 7]
```diff
@@ -709,14 +709,6 @@ packages:
...
-    dependency: transitive
-    description:
-      name: mockito
-      sha256: f99d8d072e249f719a5531735d146d8cf04c580d93920b04de75bef6dfb2daf6
-      url: "https://pub.dev"
-    source: hosted
-    version: "5.4.5"
   mocktail:
     dependency: transitive
     description:
```
Comment (ripplearcgit) [inline pubspec.lock]: can we quickly scan the usage of mocktail? 
if feasible to remove, can we add a TODO? Thanks.

**Review by ripplearcgit — APPROVED** [review]
Much more improvement!

File: pubspec.lock (hunk 7 above)
Comment (SimonDerejeW) [reply pubspec.lock]: This is a dependency brought by the `bloc_test` package, it is not used in the code base directly.
